deploy-tool/main.go

```diff
@@ -0,0 +1,224 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+// This tool consumes the output of `terraform apply`, generating keys and then
+// populating specific manifests and Kubernetes secrets with appropriate values.
+// We do this in this tool because if we generated secrets via Terraform
+// resources, the secret values would appear in the Terraform state file. The
+// struct definitions here MUST be kept in sync with the output variable in
+// terraform/modules/facilitator/facilitator.tf and the corresponding structs in
+// facilitator/src/manifest.rs.
+
+// BatchSigningPublicKey represents a public key used for batch signing.
+type BatchSigningPublicKey struct {
+	// PublicKey is the PEM armored base64 encoding of the ASN.1 encoding of the
+	// PKIX SubjectPublicKeyInfo structure. It must be an ECDSA P256 key.
+	PublicKey string `json:"public-key"`
+	// Expiration is the ISO 8601 encoded UTC date at which this key expires.
+	Expiration string `json:"expiration"`
+}
+
+// PacketEncryptionCertificate represents a certificate containing a public key
+// used for packet encryption.
+type PacketEncryptionCertificate struct {
+	// Certificate is the PEM armored X.509 certificate.
+	Certificate string `json:"certificate"`
+}
+
+// SpecificManifest represents the manifest file advertised by a data share
+// processor. See the design document for the full specification.
+// https://docs.google.com/document/d/1MdfM3QT63ISU70l63bwzTrxr93Z7Tv7EDjLfammzo6Q/edit#heading=h.3j8dgxqo5h68
+type SpecificManifest struct {
+	// Format is the version of the manifest.
+	Format float64 `json:"format"`
```

Comment:
I think at some point because it's the _Javascript_ Object Notation, I thought you had to declare the field as a float, but no, Go does the sensible thing with `int64` fields, so I changed this to `int64`.